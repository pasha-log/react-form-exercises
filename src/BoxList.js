import React, {useState} from "react";
import { v4 as uuid } from 'uuid';
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

const BoxList = () => {
    const INITIAL_STATE = [
        {id: uuid(), color: '#00FFFF', width: 100, height: 100}, 
        {id: uuid(), color: '#7FFFD4', width: 50, height: 100}
    ]
    const [boxes, setBoxes] = useState(INITIAL_STATE)
    const addBox = (newBox) => {
        setBoxes(boxes => [...boxes, {...newBox, id: uuid()}])
    }
    const deleteBox = (id) => {
        setBoxes(boxes => 
            [...boxes.filter(box => box.id !== id)]
        )
    }
    return (
        <div>
            <h3>Box List</h3>
            <NewBoxForm addBox={addBox} />
            <div>
                {boxes.map(({id, color, width, height}) => <Box id={id} key={id} color={color} width={width} height={height} deleteBox={deleteBox} />)}
            </div>

        </div>
    )
}

export default BoxList;

