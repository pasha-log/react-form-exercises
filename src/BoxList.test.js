import React from "react"; 
import {render, fireEvent} from "@testing-library/react";
import '@testing-library/jest-dom';
import BoxList from "./BoxList";

it("renders without crashing", function () {
    render(<BoxList />);
})

it("matches snapshot", function () {
    const {asFragment} = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
})

it("should add new box", function () {
    const {queryByText, getByPlaceholderText, getAllByTestId} = render(<BoxList />)
    const colorInput = getByPlaceholderText("Box Color");
    const widthInput = getByPlaceholderText("Width");
    const heightInput = getByPlaceholderText("Height");
    const btn = queryByText("Add Box");
    expect(getAllByTestId("tester").length).toBe(2);
    fireEvent.change(colorInput, {target: {value: '#8A2BE2'}})
    fireEvent.change(widthInput, {target: {value: '50'}})
    fireEvent.change(heightInput, {target: {value: '50'}})
    fireEvent.click(btn);
    expect(getAllByTestId("tester").length).toBe(3);
})