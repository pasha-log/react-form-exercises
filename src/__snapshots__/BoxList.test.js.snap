// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Box List
    </h3>
    <form>
      <label
        for="color"
      >
        Box Color:
      </label>
      <input
        id="color"
        name="color"
        placeholder="Box Color"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        placeholder="Width"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        placeholder="Height"
        type="text"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
    <div>
      <ul>
        <li>
          <div
            data-testid="tester"
            id="e391ac5a-7e0b-4b3d-a80a-5749ea04e776"
            style="background-color: rgb(0, 255, 255); width: 100px; height: 100px;"
          />
          <button>
            X
          </button>
        </li>
      </ul>
      <ul>
        <li>
          <div
            data-testid="tester"
            id="4adb26e6-3388-4bd2-91d1-a578ccd8c26e"
            style="background-color: rgb(127, 255, 212); width: 50px; height: 100px;"
          />
          <button>
            X
          </button>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
